### ISSUE TITLE
go/importer: importer.For("source") fails to load packages from testdata paths

### ISSUE THREAD

**Issue by eliasnaur** [golang/go#24392]
### What version of Go are you using (`go version`)?
go version devel +29fcd57a9f Sun Mar 4 22:53:50 2018 +0000 linux/amd64

### Does this issue reproduce with the latest release?
Yes.

### What operating system and processor architecture are you using (`go env`)?
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/elias/.cache/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/elias/dev/go"
GORACE=""
GOROOT="/home/elias/dev/go-tip"
GOTMPDIR=""
GOTOOLDIR="/home/elias/dev/go-tip/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build722960365=/tmp/go-build -gno-record-gcc-switches"

### What did you do?

Ran 
````
$ GOPATH=`pwd` go run srcimport.go
````

in the srcimporter directory from the attached [srcimporterbug.zip](https://github.com/golang/go/files/1812362/srcimporterbug.zip)

### What did you expect to see?
No errors.

### What did you see instead?
2018/03/14 18:54:52 got "." want "example.com/testdata/testpkg"

Somehow, the srcimporter is confused by the presence of "testdata" in the import path.

The importer.Default() importer works as expected, after running

````
$ GOPATH=`pwd` go install example.com/testdata/testpkg
````

I think this is related to https://github.com/golang/go/issues/23092 since its proposed [CL 83415](https://golang.org/cl/83415) also fixes this problem.

**Comment by josharian** [comment]
@griesemer

**Comment by eliasnaur** [comment]
Please consider the fix for a 1.10 point release. Between this issue and #23092, I can't use source imports from gomobile before the importer is fixed at least two releases back. Without a 1.10.1 release, that means from 1.12, almost a year from now.
Gomobile could vendor the fixed importer, but then there is the of risk version skew, as described in https://golang.org/cl/16911 where that was one of the reasons gomobile switched away from another source importer (x/tools/go/loader).

**Comment by griesemer** [comment]
@eliasnaur Filed https://github.com/golang/go/issues/24452 .
